Test failures with Python 3.14 freethreading

### Things to check first

- [x] I have searched the existing issues and didn't find my bug already reported there

- [x] I have checked that my bug is still present in the latest release


### AnyIO version

1.14.2 (c384f99687c64c59ed8a11c3a0f11a2d57daff71)

### Python version

3.14.6 freethreading

### What happened?

When running the test suite on Python 3.14 freethreading, I'm getting a few test failures. Not 100% sure if the issue's in anyio or trio, though (but trio tests pass 100%).

```pytb
py314t: install_dependency-groups> python -I -m pip install 'blockbuster>=1.5.23; python_version < "3.15"' 'coverage[toml]>=7' 'hypothesis>=4.18.2' 'psutil>=5.9' 'pytest-mock>=3.14' 'pytest-timeout>=2.4.0' 'pytest>=9.0' 'trustme>=1.0.0' 'truststore>=0.9.1' 'uvloop>=0.22.1; platform_python_implementation == "CPython" and platform_system != "Windows" and python_version < "3.15"' 'winloop>=0.2.3; platform_python_implementation == "CPython" and platform_system == "Windows"'
.pkg-cpython314t: install_requires> python -I -m pip install 'setuptools>=77' 'setuptools_scm>=6.4'
.pkg-cpython314t: _optional_hooks> python /usr/lib/python3.14/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg-cpython314t: get_requires_for_build_editable> python /usr/lib/python3.14/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg-cpython314t: build_editable> python /usr/lib/python3.14/site-packages/pyproject_api/_backend.py True setuptools.build_meta
py314t: install_package_deps> python -I -m pip install 'exceptiongroup>=1.0.2; python_version < "3.11"' 'idna>=2.8' 'trio>=0.32.0' 'typing_extensions>=4.5; python_version < "3.13"'                                                    
py314t: install_package> python -I -m pip install --force-reinstall --no-deps /tmp/anyio/.tox/.tmp/package/1/anyio-4.14.2-0.editable-py3-none-any.whl
py314t: commands[0]> coverage run -m pytest
========================================================= test session starts =========================================================
platform linux -- Python 3.14.6, pytest-9.1.1, pluggy-1.6.0
cachedir: .tox/py314t/.pytest_cache
rootdir: /tmp/anyio
configfile: pyproject.toml
testpaths: tests
plugins: anyio-4.14.2, mock-3.15.1, timeout-2.4.0
timeout: 20.0s
timeout method: signal
timeout func_only: False
collected 3996 items                                                                                                                  

tests/streams/test_buffered.py ............................................                                                     [  1%]
tests/streams/test_file.py ................................................                                                     [  2%]
tests/streams/test_memory.py .................................................................................................. [  4%]
...........                                                                                                                     [  5%]
tests/streams/test_stapled.py ................................                                                                  [  5%]
tests/streams/test_text.py ........................                                                                             [  6%]
tests/streams/test_tls.py ................................................................................                      [  8%]
tests/test_all_attributes.py .                                                                                                  [  8%]
tests/test_contextmanagers.py .................................................                                                 [  9%]
tests/test_debugging.py ........................s.......                                                                        [ 10%]
tests/test_deprecations.py .                                                                                                    [ 10%]
tests/test_eventloop.py .......................s.                                                                               [ 11%]
tests/test_fileio.py .............................................................................s............................ [ 13%]
..s...................................ssss..................................................................................... [ 16%]
........................................................ssss................................................................... [ 20%]
................................................................                                                                [ 21%]
tests/test_from_thread.py ..................................................................................................... [ 24%]
...............................................................................................ssssFFFFFFFx....                 [ 27%]
tests/test_functools.py ....................................................................................................... [ 29%]
.................................                                                                                               [ 30%]
tests/test_itertools.py ....................................................................................................... [ 33%]
............................................................................................................................... [ 36%]
............................................................................................................................... [ 39%]
.......................................                                                                                         [ 40%]
tests/test_lowlevel.py ............................................                                                             [ 41%]
tests/test_pytest_plugin.py ......................................                                                              [ 42%]
tests/test_signals.py ............                                                                                              [ 42%]
tests/test_sockets.py ......................................................................................................... [ 45%]
............................................................................................................................... [ 48%]
............................................................................................................................... [ 51%]
............................................................................................................................... [ 54%]
............................................................................................................................... [ 58%]
............................................................................................................................... [ 61%]
............................................................................................................................... [ 64%]
............................................................................................................................... [ 67%]
............................................................................................................................... [ 70%]
............................................................................................................................... [ 73%]
.....                                                                                                                           [ 74%]
tests/test_subprocesses.py ...................................................................sss..........................     [ 76%]
tests/test_synchronization.py ................................................................................................. [ 78%]
............................................................................................................................... [ 82%]
.................................                                                                                               [ 82%]
tests/test_taskgroups.py ...................................................................................................... [ 85%]
...................................................................xxxx........................................................ [ 88%]
..............................................................................ssss............................................. [ 91%]
............................................................................................................................... [ 94%]
...................                                                                                                             [ 95%]
tests/test_tempfile.py ............................................................                                             [ 96%]
tests/test_to_interpreter.py ............                                                                                       [ 97%]
tests/test_to_process.py ............................................                                                           [ 98%]
tests/test_to_thread.py ............................................................FFF.                                        [ 99%]
tests/test_typedattr.py .                                                                                                       [100%]

============================================================== FAILURES ===============================================================
_________________________________________ TestBlockingPortal.test_from_async[asyncio-asyncio] _________________________________________
tests/test_from_thread.py:780: in test_from_async
    with start_blocking_portal(portal_backend_name) as portal:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/contextlib.py:141: in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
src/anyio/from_thread.py:548: in start_blocking_portal
    portal = future.result()
             ^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:454: in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:396: in __get_result
    raise self._exception
src/anyio/from_thread.py:536: in run_blocking_portal
    run_eventloop(
src/anyio/_core/_eventloop.py:62: in run
    raise RuntimeError(f"Already running {asynclib_name} in this thread")
E   RuntimeError: Already running asyncio in this thread
__________________________________________ TestBlockingPortal.test_from_async[asyncio-trio] ___________________________________________
tests/test_from_thread.py:780: in test_from_async
    with start_blocking_portal(portal_backend_name) as portal:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/contextlib.py:141: in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
src/anyio/from_thread.py:548: in start_blocking_portal
    portal = future.result()
             ^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:454: in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:396: in __get_result
    raise self._exception
src/anyio/from_thread.py:536: in run_blocking_portal
    run_eventloop(
src/anyio/_core/_eventloop.py:62: in run
    raise RuntimeError(f"Already running {asynclib_name} in this thread")
E   RuntimeError: Already running asyncio in this thread
_____________________________________ TestBlockingPortal.test_from_async[asyncio+uvloop-asyncio] ______________________________________
tests/test_from_thread.py:780: in test_from_async
    with start_blocking_portal(portal_backend_name) as portal:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/contextlib.py:141: in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
src/anyio/from_thread.py:548: in start_blocking_portal
    portal = future.result()
             ^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:454: in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:396: in __get_result
    raise self._exception
src/anyio/from_thread.py:536: in run_blocking_portal
    run_eventloop(
src/anyio/_core/_eventloop.py:62: in run
    raise RuntimeError(f"Already running {asynclib_name} in this thread")
E   RuntimeError: Already running asyncio in this thread
_______________________________________ TestBlockingPortal.test_from_async[asyncio+uvloop-trio] _______________________________________
tests/test_from_thread.py:780: in test_from_async
    with start_blocking_portal(portal_backend_name) as portal:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/contextlib.py:141: in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
src/anyio/from_thread.py:548: in start_blocking_portal
    portal = future.result()
             ^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:454: in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:396: in __get_result
    raise self._exception
src/anyio/from_thread.py:536: in run_blocking_portal
    run_eventloop(
src/anyio/_core/_eventloop.py:62: in run
    raise RuntimeError(f"Already running {asynclib_name} in this thread")
E   RuntimeError: Already running asyncio in this thread
______________________________________ TestBlockingPortal.test_from_async[asyncio+eager-asyncio] ______________________________________
tests/test_from_thread.py:780: in test_from_async
    with start_blocking_portal(portal_backend_name) as portal:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/contextlib.py:141: in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
src/anyio/from_thread.py:548: in start_blocking_portal
    portal = future.result()
             ^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:454: in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:396: in __get_result
    raise self._exception
src/anyio/from_thread.py:536: in run_blocking_portal
    run_eventloop(
src/anyio/_core/_eventloop.py:62: in run
    raise RuntimeError(f"Already running {asynclib_name} in this thread")
E   RuntimeError: Already running asyncio in this thread
_______________________________________ TestBlockingPortal.test_from_async[asyncio+eager-trio] ________________________________________
tests/test_from_thread.py:780: in test_from_async
    with start_blocking_portal(portal_backend_name) as portal:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/contextlib.py:141: in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
src/anyio/from_thread.py:548: in start_blocking_portal
    portal = future.result()
             ^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:454: in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:396: in __get_result
    raise self._exception
src/anyio/from_thread.py:536: in run_blocking_portal
    run_eventloop(
src/anyio/_core/_eventloop.py:62: in run
    raise RuntimeError(f"Already running {asynclib_name} in this thread")
E   RuntimeError: Already running asyncio in this thread
__________________________________________ TestBlockingPortal.test_from_async[trio-asyncio] ___________________________________________
tests/test_from_thread.py:780: in test_from_async
    with start_blocking_portal(portal_backend_name) as portal:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/contextlib.py:141: in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
src/anyio/from_thread.py:548: in start_blocking_portal
    portal = future.result()
             ^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:454: in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.14t/concurrent/futures/_base.py:396: in __get_result
    raise self._exception
src/anyio/from_thread.py:536: in run_blocking_portal
    run_eventloop(
src/anyio/_core/_eventloop.py:62: in run
    raise RuntimeError(f"Already running {asynclib_name} in this thread")
E   RuntimeError: Already running trio in this thread
___________________________________________ test_run_sync_worker_cyclic_references[asyncio] ___________________________________________
tests/test_to_thread.py:386: in test_run_sync_worker_cyclic_references
    assert gc.get_referrers(contextval) == no_other_refs()
E   assert [<coroutine o...x58b73d7dc80>] == [<coroutine o...x58b800fc600>]
E     
E     Left contains one more item: <hamt_bitmap_node object at 0x58b73d7dc80>
E     Use -v to get more diff
_______________________________________ test_run_sync_worker_cyclic_references[asyncio+uvloop] ________________________________________
tests/test_to_thread.py:386: in test_run_sync_worker_cyclic_references
    assert gc.get_referrers(contextval) == no_other_refs()
E   assert [<coroutine o...x58b73d76cc0>] == [<coroutine o...x58b800f4f80>]
E     
E     Left contains one more item: <hamt_bitmap_node object at 0x58b73d76cc0>
E     Use -v to get more diff
________________________________________ test_run_sync_worker_cyclic_references[asyncio+eager] ________________________________________
tests/test_to_thread.py:386: in test_run_sync_worker_cyclic_references
    assert gc.get_referrers(contextval) == no_other_refs()
E   assert [<coroutine o...x58b73d7a740>] == [<coroutine o...x58b800f9f00>]
E     
E     Left contains one more item: <hamt_bitmap_node object at 0x58b73d7a740>
E     Use -v to get more diff
======================================================= short test summary info =======================================================
SKIPPED [1] tests/test_debugging.py:102: Generator based coroutines have been removed in Python 3.11
SKIPPED [1] tests/test_eventloop.py:132: winloop not installed
SKIPPED [1] tests/test_fileio.py:202: Drive only makes sense on Windows
SKIPPED [1] tests/test_fileio.py:247: Only makes sense on Windows
SKIPPED [4] tests/test_fileio.py:317: Could not find a suitable block device
SKIPPED [4] tests/test_fileio.py:640: os.lchmod() is not available
SKIPPED [4] tests/test_from_thread.py:745: This test is known to hang often on CI
SKIPPED [1] tests/test_subprocesses.py:371: the 'user' argument is not supported by uvloop yet
SKIPPED [1] tests/test_subprocesses.py:371: the 'group' argument is not supported by uvloop yet
SKIPPED [1] tests/test_subprocesses.py:371: the 'umask' argument is not supported by uvloop yet
SKIPPED [1] tests/test_taskgroups.py:1315: Generator based coroutines have been removed in Python 3.11
SKIPPED [3] tests/test_taskgroups.py:1336: Generator based coroutines have been removed in Python 3.11
FAILED tests/test_from_thread.py::TestBlockingPortal::test_from_async[asyncio-asyncio] - RuntimeError: Already running asyncio in this thread
FAILED tests/test_from_thread.py::TestBlockingPortal::test_from_async[asyncio-trio] - RuntimeError: Already running asyncio in this thread
FAILED tests/test_from_thread.py::TestBlockingPortal::test_from_async[asyncio+uvloop-asyncio] - RuntimeError: Already running asyncio in this thread
FAILED tests/test_from_thread.py::TestBlockingPortal::test_from_async[asyncio+uvloop-trio] - RuntimeError: Already running asyncio in this thread
FAILED tests/test_from_thread.py::TestBlockingPortal::test_from_async[asyncio+eager-asyncio] - RuntimeError: Already running asyncio in this thread
FAILED tests/test_from_thread.py::TestBlockingPortal::test_from_async[asyncio+eager-trio] - RuntimeError: Already running asyncio in this thread
FAILED tests/test_from_thread.py::TestBlockingPortal::test_from_async[trio-asyncio] - RuntimeError: Already running trio in this thread
FAILED tests/test_to_thread.py::test_run_sync_worker_cyclic_references[asyncio] - assert [<coroutine o...x58b73d7dc80>] == [<coroutine o...x58b800fc600>]
FAILED tests/test_to_thread.py::test_run_sync_worker_cyclic_references[asyncio+uvloop] - assert [<coroutine o...x58b73d76cc0>] == [<coroutine o...x58b800f4f80>]
FAILED tests/test_to_thread.py::test_run_sync_worker_cyclic_references[asyncio+eager] - assert [<coroutine o...x58b73d7a740>] == [<coroutine o...x58b800f9f00>]
================================= 10 failed, 3958 passed, 23 skipped, 5 xfailed in 231.36s (0:03:51) ==================================
py314t: exit 1 (232.77 seconds) /tmp/anyio> coverage run -m pytest pid=111401
  py314t: FAIL code 1 (248.56=setup[15.79]+cmd[232.77] seconds)
  evaluation failed :( (249.84 seconds)
```

### How can we reproduce the bug?

```
tox -e py314t
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -61,7 +61,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        python-version: ["3.10", "3.11", "3.12", "3.13", "3.14", "3.15", "pypy3.11-v7.3.23"]
+        python-version: ["3.10", "3.11", "3.12", "3.13", "3.14", "3.14t", "3.15", "pypy3.11-v7.3.23"]
         extras: ["[trio]"]
         lowest-bound: [false]
         include:

--- a/docs/versionhistory.rst
+++ b/docs/versionhistory.rst
@@ -3,6 +3,22 @@ Version history
 
 This library adheres to `Semantic Versioning 2.0 <http://semver.org/>`_.
 
+**UNRELEASED**
+
+- Added the free-threaded build of Python 3.14t to the test matrix
+  (`#1224 <https://github.com/agronholm/anyio/pull/1224>`_; PR by @EmmanuelNiyonshuti)
+  As a result:
+
+  - Fixed ``start_blocking_portal()`` portal thread inheriting the calling thread's
+    ``contextvars.Context`` on Python 3.14+ free-threaded builds, previously was
+    leading to a ``RuntimeError: Already running <backend> in this thread``
+    raised in ``TestBlockingPortal::test_from_async``.
+    (`#1220 <https://github.com/agronholm/anyio/issues/1220>`_; PR by @EmmanuelNiyonshuti)
+  - Fixed ``to_thread.run_sync()``'s worker thread pool on asyncio backend retaining a
+    reference to the calling task's ``contextvars.Context`` for the lifetime of the
+    pooled worker thread on Python 3.14+ free-threaded builds. Same reason as above.
+    previously was breaking this test ``test_run_sync_worker_cyclic_references``.
+
 **4.14.2**
 
 - Changed ``ByteReceiveStream.receive()`` implementations to raise a ``ValueError`` when

--- a/docs/versionhistory.rst
+++ b/docs/versionhistory.rst
@@ -5,7 +5,6 @@ This library adheres to `Semantic Versioning 2.0 <http://semver.org/>`_.
 
 **UNRELEASED**
 
-  (`#1224 <https://github.com/agronholm/anyio/pull/1224>`_; PR by @EmmanuelNiyonshuti)
 - Changed the default name for a task spawned with ``TaskGroup.create_task(func())`` to
   match the default task name for the analogous task spawned with
   ``TaskGroup.start_soon(func)`` or ``TaskGroup.start(func)`` in more situations.
@@ -16,6 +15,7 @@ This library adheres to `Semantic Versioning 2.0 <http://semver.org/>`_.
   free-threading builds, newly created threads inherit the current context by default,
   causing AnyIO to behave erroneously in relation to ``start_blocking_portal()`` and
   ``anyio.to_thread.run_sync()``
+  (`#1224 <https://github.com/agronholm/anyio/pull/1224>`_; PR by @EmmanuelNiyonshuti)
 
 **4.14.2**
 

--- a/docs/versionhistory.rst
+++ b/docs/versionhistory.rst
@@ -5,16 +5,11 @@ This library adheres to `Semantic Versioning 2.0 <http://semver.org/>`_.
 
 **UNRELEASED**
 
-- Added the free-threaded build of Python 3.14t to the test matrix
+- Fixed free-threading compatibility issues arising from the fact that on Python 3.14
+  free-threading builds, newly created threads inherit the current context by default,
+  causing AnyIO to behave erroneously in relation to ``start_blocking_portal()`` and
+  ``anyio.to_thread.run_sync()``
   (`#1224 <https://github.com/agronholm/anyio/pull/1224>`_; PR by @EmmanuelNiyonshuti)
-  As a result:
-
-  - Fixed ``start_blocking_portal()`` portal thread inheriting the calling thread's
-    ``contextvars.Context`` on Python 3.14+ free-threaded build.
-    (`#1220 <https://github.com/agronholm/anyio/issues/1220>`_; PR by @EmmanuelNiyonshuti)
-  - Fixed ``to_thread.run_sync()``'s worker thread pool on asyncio backend retaining a
-    reference to the calling task's ``contextvars.Context`` for the lifetime of the
-    pooled worker thread on Python 3.14+ free-threaded build.
 
 **4.14.2**
 

--- a/docs/versionhistory.rst
+++ b/docs/versionhistory.rst
@@ -10,14 +10,11 @@ This library adheres to `Semantic Versioning 2.0 <http://semver.org/>`_.
   As a result:
 
   - Fixed ``start_blocking_portal()`` portal thread inheriting the calling thread's
-    ``contextvars.Context`` on Python 3.14+ free-threaded builds, previously was
-    leading to a ``RuntimeError: Already running <backend> in this thread``
-    raised in ``TestBlockingPortal::test_from_async``.
+    ``contextvars.Context`` on Python 3.14+ free-threaded build.
     (`#1220 <https://github.com/agronholm/anyio/issues/1220>`_; PR by @EmmanuelNiyonshuti)
   - Fixed ``to_thread.run_sync()``'s worker thread pool on asyncio backend retaining a
     reference to the calling task's ``contextvars.Context`` for the lifetime of the
-    pooled worker thread on Python 3.14+ free-threaded builds. Same reason as above.
-    previously was breaking this test ``test_run_sync_worker_cyclic_references``.
+    pooled worker thread on Python 3.14+ free-threaded build.
 
 **4.14.2**
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -24,6 +24,7 @@ classifiers = [
     "Programming Language :: Python :: 3.13",
     "Programming Language :: Python :: 3.14",
     "Programming Language :: Python :: 3.15",
+    "Programming Language :: Python :: Free Threading :: 2 - Beta",
 ]
 requires-python = ">= 3.10"
 dependencies = [

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -154,7 +154,7 @@ exclude_also = [
 ]
 
 [tool.tox]
-env_list = ["pre-commit", "py310", "py311", "py312", "py313", "py314", "py315", "pypy3", "only_asyncio"]
+env_list = ["pre-commit", "py310", "py311", "py312", "py313", "py314", "py314t", "py315", "pypy3", "only_asyncio"]
 skip_missing_interpreters = true
 requires = ["tox >= 4.22"]
 

--- a/src/anyio/_backends/_asyncio.py
+++ b/src/anyio/_backends/_asyncio.py
@@ -991,7 +991,7 @@ class WorkerThread(Thread):
     ):
         kwargs: dict[str, Any] = {}
         if sys.version_info >= (3, 14):
-            kwargs["context"] = False
+            kwargs["context"] = Context()
 
         super().__init__(name="AnyIO worker thread", **kwargs)
         self.root_task = root_task

--- a/src/anyio/_backends/_asyncio.py
+++ b/src/anyio/_backends/_asyncio.py
@@ -989,7 +989,7 @@ class WorkerThread(Thread):
     ):
         kwargs: dict[str, Any] = {}
         if sys.version_info >= (3, 14):
-            kwargs["context"] = Context()
+            kwargs["context"] = False
 
         super().__init__(name="AnyIO worker thread", **kwargs)
         self.root_task = root_task

--- a/src/anyio/_backends/_asyncio.py
+++ b/src/anyio/_backends/_asyncio.py
@@ -987,7 +987,10 @@ class WorkerThread(Thread):
         workers: set[WorkerThread],
         idle_workers: deque[WorkerThread],
     ):
-        super().__init__(name="AnyIO worker thread")
+        kwargs: dict[str, Any] = {}
+        if sys.version_info >= (3, 14):
+            kwargs["context"] = Context()
+        super().__init__(name="AnyIO worker thread", **kwargs)
         self.root_task = root_task
         self.workers = workers
         self.idle_workers = idle_workers

--- a/src/anyio/_backends/_asyncio.py
+++ b/src/anyio/_backends/_asyncio.py
@@ -990,6 +990,7 @@ class WorkerThread(Thread):
         kwargs: dict[str, Any] = {}
         if sys.version_info >= (3, 14):
             kwargs["context"] = Context()
+
         super().__init__(name="AnyIO worker thread", **kwargs)
         self.root_task = root_task
         self.workers = workers

--- a/src/anyio/from_thread.py
+++ b/src/anyio/from_thread.py
@@ -545,6 +545,7 @@ def start_blocking_portal(
     kwargs: dict[str, Any] = {}
     if sys.version_info >= (3, 14):
         kwargs["context"] = Context()
+
     thread = Thread(target=run_blocking_portal, daemon=True, name=name, **kwargs)
     thread.start()
     try:

--- a/src/anyio/from_thread.py
+++ b/src/anyio/from_thread.py
@@ -9,7 +9,6 @@ __all__ = (
     "start_blocking_portal",
 )
 
-import contextvars
 import sys
 from collections.abc import Awaitable, Callable, Coroutine, Generator
 from concurrent.futures import Future
@@ -18,6 +17,7 @@ from contextlib import (
     AbstractContextManager,
     contextmanager,
 )
+from contextvars import Context
 from dataclasses import dataclass, field
 from functools import partial
 from inspect import isawaitable
@@ -542,15 +542,10 @@ def start_blocking_portal(
                     future.set_exception(exc)
 
     future: Future[BlockingPortal] = Future()
+    kwargs: dict[str, Any] = {}
     if sys.version_info >= (3, 14):
-        thread = Thread(
-            target=run_blocking_portal,
-            daemon=True,
-            name=name,
-            context=contextvars.Context(),
-        )
-    else:
-        thread = Thread(target=run_blocking_portal, daemon=True, name=name)
+        kwargs["context"] = Context()
+    thread = Thread(target=run_blocking_portal, daemon=True, name=name, **kwargs)
     thread.start()
     try:
         cancel_remaining_tasks = False

--- a/src/anyio/from_thread.py
+++ b/src/anyio/from_thread.py
@@ -17,6 +17,7 @@ from contextlib import (
     AbstractContextManager,
     contextmanager,
 )
+from contextvars import Context
 from dataclasses import dataclass, field
 from functools import partial
 from inspect import isawaitable
@@ -543,8 +544,7 @@ def start_blocking_portal(
     future: Future[BlockingPortal] = Future()
     kwargs: dict[str, Any] = {}
     if sys.version_info >= (3, 14):
-        kwargs["context"] = False
-
+        kwargs["context"] = Context()
     thread = Thread(target=run_blocking_portal, daemon=True, name=name, **kwargs)
     thread.start()
     try:

--- a/src/anyio/from_thread.py
+++ b/src/anyio/from_thread.py
@@ -9,6 +9,7 @@ __all__ = (
     "start_blocking_portal",
 )
 
+import contextvars
 import sys
 from collections.abc import Awaitable, Callable, Coroutine, Generator
 from concurrent.futures import Future
@@ -541,7 +542,15 @@ def start_blocking_portal(
                     future.set_exception(exc)
 
     future: Future[BlockingPortal] = Future()
-    thread = Thread(target=run_blocking_portal, daemon=True, name=name)
+    if sys.version_info >= (3, 14):
+        thread = Thread(
+            target=run_blocking_portal,
+            daemon=True,
+            name=name,
+            context=contextvars.Context(),
+        )
+    else:
+        thread = Thread(target=run_blocking_portal, daemon=True, name=name)
     thread.start()
     try:
         cancel_remaining_tasks = False

--- a/src/anyio/from_thread.py
+++ b/src/anyio/from_thread.py
@@ -17,7 +17,6 @@ from contextlib import (
     AbstractContextManager,
     contextmanager,
 )
-from contextvars import Context
 from dataclasses import dataclass, field
 from functools import partial
 from inspect import isawaitable
@@ -544,7 +543,8 @@ def start_blocking_portal(
     future: Future[BlockingPortal] = Future()
     kwargs: dict[str, Any] = {}
     if sys.version_info >= (3, 14):
-        kwargs["context"] = Context()
+        kwargs["context"] = False
+
     thread = Thread(target=run_blocking_portal, daemon=True, name=name, **kwargs)
     thread.start()
     try:

--- a/tests/test_to_thread.py
+++ b/tests/test_to_thread.py
@@ -24,6 +24,7 @@ from anyio import (
 )
 from anyio._core._eventloop import current_async_library
 from anyio.from_thread import BlockingPortalProvider
+from anyio.lowlevel import checkpoint
 
 from .conftest import asyncio_params, no_other_refs
 
@@ -382,6 +383,11 @@ async def test_run_sync_worker_cyclic_references() -> None:
     await to_thread.run_sync(foo, arg)
     cvar.set(Foo())
     gc.collect()
+    # This checkpoint only exist to give the worker thread a moment to finish releasing its own
+    # references before we check. on free threaded python builds, this can
+    # otherwise race and intermittently show a leftover reference.
+    # See: https://github.com/agronholm/anyio/pull/1224
+    await checkpoint()
 
     assert gc.get_referrers(contextval) == no_other_refs()
     assert gc.get_referrers(foo) == no_other_refs()

--- a/tests/test_to_thread.py
+++ b/tests/test_to_thread.py
@@ -383,10 +383,6 @@ async def test_run_sync_worker_cyclic_references() -> None:
     await to_thread.run_sync(foo, arg)
     cvar.set(Foo())
     gc.collect()
-    # This checkpoint only exist to give the worker thread a moment to finish releasing its own
-    # references before we check. on free threaded python builds, this can
-    # otherwise race and intermittently show a leftover reference.
-    # See: https://github.com/agronholm/anyio/pull/1224
     await checkpoint()
 
     assert gc.get_referrers(contextval) == no_other_refs()

